python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -18,9 +18,93 @@
 from cudf.utils.dtypes import is_categorical_dtype
 
 
-def set_partitions_hash(df, columns, npartitions):
-    c = hash_object_dispatch(df[columns], index=False).values
-    return np.mod(c, npartitions)
+class All2All:
+    def __init__(
+        self, inputs, columns, npartitions, k, n, stages, token, ignore_index
+    ):
+        self._all2all = True
+        self.npartitions = npartitions
+        self.n = n
+        self.k = k
+        self.stages = stages
+        self.inputs = inputs  # Input IDs
+        self.token = token
+        self.columns = columns
+        self.ignore_index = ignore_index
+        self.output_keys = []
+
+    def __repr__(self):
+        return (
+            f"All2All(input_keys={self.input_keys}, "
```

Comment:
@rjzamora can we remove this reference to `self.input_keys` unless we need the keys themselves as part of the repr? It currently doesn't exist in the class